htdocs/categories/class/categorie.class.php

```diff
@@ -753,6 +753,7 @@ function getObjectsInCateg($type)
 		$sql .= " WHERE o.entity IN (" . getEntity( $obj->element, 1).")";
 		$sql.= " AND c.fk_categorie = ".$this->id;
 		$sql .= " AND c.fk_" . $this->MAP_CAT_FK[$type] . " = o.rowid";
+		if ($field=='societe') $sql.= " ORDER by o.nom ASC";
```

Comment:
Le 18. 02. 16 00:49, Laurent Destailleur a écrit :
>
> In htdocs/categories/class/categorie.class.php
> <https://github.com/Dolibarr/dolibarr/pull/4644#discussion_r53251625>:
>
> > @@ -753,6 +753,7 @@ function getObjectsInCateg($type)
> >  		$sql .= " WHERE o.entity IN (" . getEntity( $obj->element, 1).")";
> >  		$sql.= " AND c.fk_categorie = ".$this->id;
> >  		$sql .= " AND c.fk_" . $this->MAP_CAT_FK[$type] . " = o.rowid";
> > +		if ($field=='societe') $sql.= " ORDER by o.nom ASC";
>
> Variable $field is not defined. Also it would be better to sort after
> reception of result by the caller of function.
> Having a sort criteria should be better into function that use the result.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Dolibarr/dolibarr/pull/4644/files#r53251625>.
>
Vous avez raison, mais je ne suis pas sûr de savoir comment le faire.
Et si les critères de tri pouvaient être accessibles à l'utilisateur, ce
serait top.